### PR TITLE
Spanish User Facing Fixes

### LINKED ISSUE #1633: Final Pass - User Facing Spanish Review
This ticket is to capture the last needed Spanish translations _for the Case Manager facing sections_. Admin & Data manager sections _are not required_ for this review.

* **What are we trying to do?**

Browse around on the [Spanish Review](https://github.com/DCAFEngineering/dcaf_case_management/tree/spanish_review) branch and look for English. Look at the regular text, any helper text (the "?"s), error messages, drop down selections. Try clicking and editing things. Sort things. Edit the call list. Try to break things!

Do not worry about the user entered text entered in fields. Values in notes, the patient name itself, other contact relationship, etc are not what we're looking for here. If you're uncertain though, just go ahead and add a comment of it! 

Screen shot, or carefully describe in a comment here, the English.

Also look out for any errors related to translations.

* **What feature or behavior is this required for?**

User facing Spanish release!

* **How could we solve this issue? (Not knowing is okay!)**

Lotsa eyes.

### PR DESCRIPTION
**Pull request by lomky**
I rule and have completed some work on Case Manager that's ready for review!

PR for some changes from #1633 

This pull request makes the following changes:
* Login page "Forgot your password" translated
* Forgot your Password page translated.
* Add Spanish/English button on login page
* Fixes login page related alert messages
* Calls other contact case translation

It relates to the following issue #s: 
* Fixes #1633

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -6,13 +6,19 @@
 
     <p><%= t('devise.sessions.new.password_signin') %></p>
 
+    <!--
```
Comment (colinxfleming) [inline app/views/devise/sessions/new.html.erb]: Let's make this a ERB ruby comment instead of an html comment -- `<%# whatever %>

Comment (lomky) [reply app/views/devise/sessions/new.html.erb]: it wasn't seeming to fool the i18n tasks script as ERB ruby, but I'll give it another go!

Comment (colinxfleming) [reply app/views/devise/sessions/new.html.erb]: oh word -- in that case, stick these in the Devise controller or Oauth controller as comments, per https://github.com/glebm/i18n-tasks#fine-tuning

File: config/locales/es.yml [hunk 2]
```diff
@@ -106,13 +110,27 @@ es:
   devise:
     omniauth_callbacks:
       success: Iniciar sesión con éxito con %{kind}
+    password:
+      new:
+        send_reset: Enviarme las instrucciones para restablecer la contraseña
```
Comment (lomky) [inline config/locales/es.yml]:
```suggestion
        send_reset: Enviame las instrucciones para restablecer mi contraseña
```

File: config/locales/es.yml [hunk 3]
```diff
@@ -173,7 +191,9 @@ es:
         phone: Teléfono
         state: Estado
         street_address: Dirección
-        zip: Código postal
+        zip: ZIP
```
Comment (lomky) [inline config/locales/es.yml]:
```suggestion
        zip: Código postal```

**Pull request merged**